### PR TITLE
Port all_builds.yml to be compatible to GDExtension

### PR DESCRIPTION
**Pull request by 2shady4u**
@bitbrain 

This ports Github Actions compilation for all platforms to be compatible with the new GDExtension.
I'm still having some issues with iOS, but I imagine it's related to FMOD having to update their binaries (?).

I'll see if I can port the automatic GUT and release stage as well at some point.

EDIT: I propose to merge this branch as soon as possible into godot-4.0-adaption such that we can detect any new compilation bugs and errors immediately and effortlessly.

### CODE REVIEW

File: .github/workflows/all_builds.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: 🌈 All Builds
 on:
   push:
-    branches: [ master ]
+    branches: [ master, godot-4.0-adaption ]
```
Comment (bitbrain) [inline .github/workflows/all_builds.yml:4]: could we also add godot-4.0 here, too?

File: src/callback/file_callbacks.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #include "../helpers/containers.h"
 #include <thread>
 #include <condition_variable>
+#include <cstring> // This include is required for both Linux and MacOS targets as they don't include the necessary headers for 'memcpy' by default
```
Comment (bitbrain) [inline src/callback/file_callbacks.h:10]: 👀

Comment (2shady4u) [reply src/callback/file_callbacks.h:10]: Classical Linux/MacOS C++ thing 🤷 
I left a comment such that it is clear that this is not required for Windows.

File: .github/workflows/all_builds.yml (hunk 1 above)
Comment (2shady4u) [reply .github/workflows/all_builds.yml:4]: Sure, I'll add it.
When godot-4.0-adaption gets merged, this will be cleaned up most likely.

Comment (bitbrain) [reply .github/workflows/all_builds.yml]: Correct!

File: src/callback/file_callbacks.h (hunk 2 above)
Comment (bitbrain) [reply src/callback/file_callbacks.h:10]: I think it is good as it is.

**Pull request merged**